### ISSUE TITLE
train 30 epochs on coco but get no detections

### ISSUE THREAD

**Issue by zhangliukun** [eriklindernoren/PyTorch-YOLOv3#111]
I train on the coco according the code for 30 epochs,the result is as follows:
[Epoch 29/30, Batch 7328/7329] [Losses: x 0.141817, y 0.112415, w 0.084596, h 0.050634, conf 0.269597, cls 0.676112, total 1.335171, recall: 0.41742, precision: 0.07105]

When I perform detect.py through the trained 29.weights instead of  the yolov3.weights, all the output pictures have no results such as bounding box. Has anybody else encountered this problem?

**Comment by Ooshita** [comment]
The number of epochs 30 is too small.
Since the accuracy was only 0.07 = 7%, I think that the bounding box was not detected.

**Comment by tamasbalassa** [comment]
Yeah, I've had the same problem. Seems like the training part of this project is not yet finished (/working properly).

**Comment by quick-composite** [comment]
@tamasbalassa @Ooshita @zhangliukun this is a good PyTorch YOLOv3 repository. This trains on COCO from scratch to similar results as darknet. You need to train for about 60-70 epochs to achieve full results, but you can see from the repository readme that even after 30 epochs the mAP is about 0.50 already:
https://github.com/ultralytics/yolov3

![](https://user-images.githubusercontent.com/26833433/49822374-3b27bf00-fd7d-11e8-9180-f0ac9fe2fdb4.png)

**Comment by ming71** [comment]
> s link below is a good PyTorch YOLOv3 repository. This trains on COCO from scratch to similar results as darknet. You need to train for about 60-70 epochs to achieve full results, but you can see from the repository read

Taking your advice to use another repo,
But why I always failed to train on my own datasets?
According to the describtion , COCO format label should be used to image , but I found it not okay , its code shows that normalized input labels are required instead of coco format!!!why?
(refer to my descrebtion here : [ultralytics/yolov3#95](https://github.com/ultralytics/yolov3/issues/95))

**Comment by fourth-archive** [comment]
@ming71 the repo has a tutorial for training custom data that should answer all your questions:
https://github.com/ultralytics/yolov3/wiki/Example:-Train-Single-Class

**Comment by ming71** [comment]
> @ming71 the repo

Thanks for your reply.I have found my fault weeks ago,the unmatched version leads to the problem

**Comment by BCWang93** [comment]
> I train on the coco according the code for 30 epochs,the result is as follows:
> [Epoch 29/30, Batch 7328/7329] [Losses: x 0.141817, y 0.112415, w 0.084596, h 0.050634, conf 0.269597, cls 0.676112, total 1.335171, recall: 0.41742, precision: 0.07105]
> 
> When I perform detect.py through the trained 29.weights instead of the yolov3.weights, all the output pictures have no results such as bounding box. Has anybody else encountered this problem?

Hi，I have same problem like you.Do you solved this problem?Can you help me?Thanks!

**Comment by BCWang93** [comment]
> Yeah, I've had the same problem. Seems like the training part of this project is not yet finished (/working properly).

Hi,do you solved this problem?I have the same problem like this!The detection output is not have bbox,it's only have origin image!Can you help me to solve it?Thanks!

**Comment by eriklindernoren** [comment]
The issue was in the `Darknet.save_weights` method. If you use current master it should work better.
